OpenDreamRuntime.Tests/UnitTest1.cs

```diff
@@ -444,5 +445,30 @@ public void UnicodeProcsTest()
 
             Assert.Zero(runtime.ExceptionCount);
         }
+
+        [TestCase("Hello, World!", ", ", -1, 6)]
+        [TestCase("Hello, World!", ", ", 3, 3)]
+        [TestCase("Hello, World!", ", ", 7, 11)]
+        public void NonspantextTest(string haystack, string needles, int start, int valueResult)
+        {
+            var runtime = CreateRuntime();
+
+            var haystackDreamValue = new DreamValue(haystack);
+            var needlesDreamValue = new DreamValue(needles);
+            var startDreamValue = new DreamValue(start);
+            var valueResultDreamValue = new DreamValue(valueResult);
+
+            var listDreamValue = new List<DreamValue>() { haystackDreamValue, needlesDreamValue, startDreamValue };
+            var dicDreamValue = new Dictionary<string, DreamValue>() { ["Haystack"] = haystackDreamValue, ["Needles"] = needlesDreamValue, ["Start"] = startDreamValue };
+
+            var result = DreamThread.Run(runtime, async state =>
+            {
+                var world = runtime.WorldInstance;
+                var proc = world.GetProc("nonspantext");
+                return await state.Call(proc, world, null, new DreamProcArguments(listDreamValue, dicDreamValue));
```

Comment:
`dicDreamValue` is unnecessary because the arguments are already being passed in with `listDreamValue`
```suggestion

            var result = DreamThread.Run(runtime, async state =>
            {
                var world = runtime.WorldInstance;
                var proc = world.GetProc("nonspantext");
                return await state.Call(proc, world, null, new DreamProcArguments(listDreamValue));
```